unbxdAutosuggest.js

```diff
@@ -889,21 +890,28 @@ var unbxdAutoSuggestFunction = function ($, Handlebars, undefined) {
 			var host_path = this.getHostNPath();
 
 			var url = "q=" + encodeURIComponent(this.params.q);
+
 			if (this.options.maxSuggestions) {
 				url += '&inFields.count=' + this.options.maxSuggestions
 					+ '&topQueries.count=' + this.options.maxSuggestions
 					+ '&keywordSuggestions.count=' + this.options.maxSuggestions
-					+ '&popularProducts.count=' + this.options.popularProducts.count;
+					+ '&popularProducts.count=' + this.options.popularProducts.count					
 				+ '&indent=off';
 			}
 			else {
 				url += '&inFields.count=' + this.options.inFields.count
 					+ '&topQueries.count=' + this.options.topQueries.count
 					+ '&keywordSuggestions.count=' + this.options.keywordSuggestions.count
-					+ '&popularProducts.count=' + this.options.popularProducts.count;
+					+ '&popularProducts.count=' + this.options.popularProducts.count
 				+ '&indent=off';
 			}
 
+			if (this.options.popularProducts.fields.length > 0) {
+				var popular_product_fields = this.options.popularProducts.fields.join(",");
```

Comment:
What if `self.options.popularProducts.field` is `undefined`

Reply:
it is included in default options (fields:[]) hence it won't be undefined but length 0